src/antsibull_nox/collection.py

```diff
@@ -0,0 +1,432 @@
+        v = data.get("version")
+        d = data.get("dependencies")
+    elif manifest_json.is_file():
+        found = manifest_json
+        with open(manifest_json, "br") as f:
+            data = f.read()
+        ci = data["collection_info"]
+        ns = ci["namespace"]
+        n = ci["name"]
+        v = ci.get("version")
+        d = ci.get("dependencies")
+    else:
+        raise ValueError(f"Cannot find galaxy.yml or MANIFEST.json in {path}")
+
+    if not isinstance(ns, str):
+        raise ValueError(f"{found} does not contain a namespace")
+    if not isinstance(n, str):
+        raise ValueError(f"{found} does not contain a name")
+    if not isinstance(v, str):
+        v = None
+    d = d or {}
+    if not isinstance(d, dict):
+        raise ValueError(f"{found}'s dependencies is not a mapping")
+
+    if namespace is not None and ns != namespace:
+        raise ValueError(
+            f"{found} contains namespace {ns!r}, but was hoping for {namespace!r}"
+        )
+    if name is not None and n != name:
+        raise ValueError(f"{found} contains name {n!r}, but was hoping for {name!r}")
+    return CollectionData(
+        collections_root_path=root,
+        path=path,
+        namespace=ns,
+        name=n,
+        full_name=f"{ns}.{n}",
+        version=v,
+        dependencies=d,
+        current=current,
+    )
+
+
+def _fs_list_local_collections() -> Iterator[CollectionData]:
+    root: Path | None = None
+
+    # Determine potential root
+    cwd = Path.cwd()
+    parents: Sequence[Path] = cwd.parents
+    if len(parents) > 3 and parents[2].name == "ansible_collections":
+        root = parents[3]
+
+    # Current collection
+    try:
+        current = _load_collection_data_from_disk(cwd, root=root, current=True)
+        if (
+            root
+            and current.namespace == parents[1].name
+            and current.name == parents[0].name
+        ):
+            yield current
+        else:
+            root = None
+            current = _load_collection_data_from_disk(cwd, current=True)
+            yield current
+    except Exception as exc:
+        raise ValueError(
+            f"Cannot load current collection's info from {cwd}: {exc}"
+        ) from exc
+
+    # Search tree
+    if root:  # pylint: disable=too-many-nested-blocks
+        for namespace in root.iterdir():
+            try:
+                if namespace.is_dir() or namespace.is_symlink():
+                    for name in namespace.iterdir():
+                        try:
+                            if name.is_dir() or name.is_symlink():
+                                yield _load_collection_data_from_disk(
+                                    name,
+                                    namespace=namespace.name,
+                                    name=name.name,
+                                    root=root,
+                                )
+                        except:  # noqa: E722, pylint: disable=bare-except
```

Comment:
```suggestion
                        except Exception:
```

at least, otherwise, this will catch KeyboardInterrupt and such. Or should this only catch ValueError because _load_collection_data_from_disk doesn't (directly) raise anything else?

(This also applies to the other callers of _load_collection_from_disk)

Reply:
It should also catch I/O errors if someone did something strange in that directory structure so that some other call fails.